### PR TITLE
[5194] Link to trainees from dead job viewer

### PR DESCRIPTION
**Pull request by richardpattinson**
### Context

Link to trainees from the show page of dead job viewer.

### Changes proposed in this pull request

Add a button to link to the trainee from the dead job viewer.

### Important business

- [ ] Does this PR introduce any PII fields that need to be overwritten or deleted in db/scripts/sanitise.sql?
- [ ] Does this PR change the database schema? If so, have you updated the config/analytics.yml file and considered whether you need to send 'import_entity' events?
- [ ] Do we need to send any updates to DQT as part of the work in this PR?
- [ ] Does this PR need an ADR?

NB: Please notify the #twd_data_insights team and ask for a review if new fields are being added to analytics.yml

### CODE REVIEW

**Review by d-a-v-e — CHANGES_REQUESTED** [review]
That currently will through a syntax error, try this:

```erb
<table class="govuk-table">
  <caption class="govuk-table__caption govuk-table__caption--m"><%= dead_job_service.name %></caption>
  <thead class="govuk-table__head">
    <tr class="govuk-table__row">
      <% dead_job_service.headers&.each do |header| %>
        <th scope="col" class="govuk-table__header"><%= header.to_s.humanize %></th>
      <% end %>
      <th scope="col" class="govuk-table__header"></th>
    </tr>
  </thead>
  <tbody class="govuk-table__body">
    <% dead_job_service.rows&.each do |row| %>
      <tr class="govuk-table__row">
        <% row.each do |data| %>
          <td class="govuk-table__cell"><%= data[-1] %></td>
        <% end %>
        <td class="govuk-table__cell">
          <%= render GovukButtonLinkTo::View.new(
            body: "View",
            url: trainee_path(Trainee.find(row[:register_id])),
            class_option: "govuk-!-margin-bottom-0",
          ) %>
        </td>
      </tr>
    <% end %>
  </tbody>
</table>
```

**Comment by jhenderson** [comment]
[Dead Job Viewer: Link to trainees from Show page](https://trello.com/c/UmeGCx6O/5194-dead-job-viewer-link-to-trainees-from-show-page)

File: app/views/system_admin/dead_jobs/index.html.erb [hunk 1]
```diff
@@ -30,8 +30,9 @@
               <%= render GovukButtonLinkTo::View.new(
                 body: "View",
                 url: dead_job_path(service.class),
+                id: "view_#{service.name.gsub(" ", "_").downcase}",
```
Comment (d-a-v-e) [inline app/views/system_admin/dead_jobs/index.html.erb:33]: each service has a method that looks like it does this just call `service.identifier`

Comment (d-a-v-e) [reply app/views/system_admin/dead_jobs/index.html.erb:33]: although you'll need to move the method out of private (in `dead_jobs/base.rb`)

**Review by d-a-v-e — APPROVED** [review]
make use of the `service.identifier` method then this looks G2G 👍

**Pull request merged**